### PR TITLE
Have Categorical ops defer to DataFrame; broken off of #24282

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/arrays/categorical/test_operators.py [hunk 1]
```diff
@@ -113,9 +114,34 @@ def test_comparisons(self):
...
 class TestCategoricalOps(object):
 
+    def test_compare_frame(self):
+        # GH#24282 check that Categorical.__cmp__(DataFrame) defers to frame
+        data = ["a", "b", 2, "a"]
+        cat = Categorical(data)
+
+        df = DataFrame(cat)
+
+        for op in [operator.eq, operator.ne, operator.ge,
```
Comment (jreback) [inline pandas/tests/arrays/categorical/test_operators.py]: we have a fixture (not sure if u r using it elsewhere)

**Review by jreback — CHANGES_REQUESTED** [review]
need a whatsnew note (can put with one of the other notes in ops and alignment) - does this close anything?

File: pandas/tests/arrays/categorical/test_operators.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/arrays/categorical/test_operators.py]: Doesn’t make sense to use In this context

**Comment by jbrockmendel** [comment]
Will add what’s new and check on issues

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**